framework/src/problems/DumpObjectsProblem.C

```diff
@@ -30,24 +33,57 @@ DumpObjectsProblem::validParams()
       true,
       "Whether to include all parameters that have been specified by a user in the dump, even if "
       "they match the default value of the parameter in the Factory");
+
+  // Change the default because any complex solve or executioners needs the problem to perform its
+  // setup duties (all the calls in initialSetup()) which are skipped by the DumpObjectsProblem
+  params.addParam<bool>(
+      "solve",
+      false,
+      "Whether to attempt to solve the Problem. This will only cause additional outputs of the "
+      "objects and their parameters. This is unlikely to succeed with more complex executioners.");
   return params;
 }
 
 DumpObjectsProblem::DumpObjectsProblem(const InputParameters & parameters)
   : FEProblemBase(parameters),
-    _nl_sys(std::make_shared<DumpObjectsNonlinearSystem>(*this, "nl0")),
     _include_all_user_specified_params(getParam<bool>("include_all_user_specified_params"))
 {
-  _nl[0] = _nl_sys;
-  _solver_systems[0] = std::dynamic_pointer_cast<SolverSystem>(_nl[0]);
+  // Make dummy systems based on parameters passed
+  _solver_systems.resize(0);
+  for (const auto i : index_range(_nl_sys_names))
+  {
+    const auto & sys_name = _nl_sys_names[i];
+    const auto & new_sys = std::make_shared<DumpObjectsNonlinearSystem>(*this, sys_name);
+    _solver_systems.push_back(new_sys);
+    _nl[i] = new_sys;
+  }
+  for (const auto i : index_range(_linear_sys_names))
+  {
+    const auto & sys_name = _linear_sys_names[i];
+    const auto & new_sys = std::make_shared<DumpObjectsLinearSystem>(*this, sys_name);
+    _solver_systems.push_back(new_sys);
+    _linear_systems[i] = new_sys;
+  }
   _aux = std::make_shared<AuxiliarySystem>(*this, "aux0");
+
+  // Create a dummy assembly for the systems at hand
   newAssemblyArray(_solver_systems);
 
   // Create extra vectors and matrices if any
   createTagVectors();
 
   // Create extra solution vectors if any
   createTagSolutions();
+
+  // Add an action to call printObjects at the end of the action/tasks phase
+  // NOTE: We previously relied on problem.solve() but some executioners (SIMPLE in NavierStokes) do
+  // not support this
+  auto action_params = _app.getActionFactory().getValidParams("DumpObjectsAction");
+  action_params.applyParameters(parameters);
+  // action_params.finalize("");
+  auto dump_objects_action =
+      _app.getActionFactory().create("DumpObjectsAction", "dump_objects", action_params);
+  _app.actionWarehouse().addActionBlock(dump_objects_action);
```

Comment:
It seems a little weird to have the action added by the `MooseObject`. You already have the `dynamic_cast` check in the action. Why not just always add the action then and don't worry about having this code?

At a minimum if this problem adds the action, we should have no need for the `dynamic_cast` check in the action (could assert instead)

Reply:
works for me

Reply:
I like creating the action here so I ll remove the cast. 
I think the "Problem" pertains to the setup as well and adding an action is a setup action so it should be fine